modules/base/rendering/renderabletimevaryingsphere.cpp

```diff
@@ -0,0 +1,554 @@
+                adjustedOpacity = 0.f;
+            }
+        }
+
+        if (_fadeOutThreshold > -1.0) {
+            const float logDistCamera = glm::log(static_cast<float>(
+                glm::distance(data.camera.positionVec3(), data.modelTransform.translation)
+            ));
+            const float startLogFadeDistance = glm::log(_size * _fadeOutThreshold);
+            const float stopLogFadeDistance = startLogFadeDistance + 1.f;
+
+            if (logDistCamera > startLogFadeDistance && 
+                logDistCamera < stopLogFadeDistance)
+            {
+                const float fadeFactor = glm::clamp(
+                    (logDistCamera - startLogFadeDistance) /
+                        (stopLogFadeDistance - startLogFadeDistance),
+                    0.f,
+                    1.f
+                );
+                adjustedOpacity *= (1.f - fadeFactor);
+            }
+            else if (logDistCamera >= stopLogFadeDistance) {
+                adjustedOpacity = 0.f;
+            }
+        }
+    }
+    // Performance wise
+    if (adjustedOpacity < 0.01f) {
+        return;
+    }
+
+    _shader->setUniform(_uniformCache.opacity, adjustedOpacity);
+    _shader->setUniform(_uniformCache._mirrorTexture, _mirrorTexture);
+
+    ghoul::opengl::TextureUnit unit;
+    unit.activate();
+    _texture->bind();
+    _shader->setUniform(_uniformCache.colorTexture, unit);
+
+    // Setting these states should not be necessary,
+    // since they are the default state in OpenSpace.
+    glEnable(GL_CULL_FACE);
+    glCullFace(GL_BACK);
+
+    if (orientation == Orientation::Inside) {
+        glCullFace(GL_FRONT);
+    }
+    else if (orientation == Orientation::Both) {
+        glDisable(GL_CULL_FACE);
+    }
+
+    if (_useAdditiveBlending) {
+        glBlendFunc(GL_SRC_ALPHA, GL_ONE);
+        glDepthMask(false);
+    }
+
+    _sphere->render();
+
+    if (_useAdditiveBlending) {
+        glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+        glDepthMask(true);
+    }
+
+    _shader->setIgnoreUniformLocationError(ghoul::opengl::ProgramObject::IgnoreError::No);
+    _shader->deactivate();
+
+    if (orientation == Orientation::Inside) {
+        glCullFace(GL_BACK);
+    }
+    else if (orientation == Orientation::Both) {
+        glEnable(GL_CULL_FACE);
+    }
+    glDisable(GL_CULL_FACE);
+}
+
+void RenderableTimeVaryingSphere::extractMandatoryInfoFromSourceFolder() {
+    // Ensure that the source folder exists and then extract
+    // the files with the same extension as <inputFileTypeString>
+    namespace fs = std::filesystem;
+    fs::path sourceFolder = absPath(_textureSourcePath);
+    if (!std::filesystem::is_directory(sourceFolder)) {
+        throw ghoul::RuntimeError(
+            "Source folder for timevaryingsphere is not a valid directory"
+        );
+    }
+    // Extract all file paths from the provided folder
+    _files.clear();
+    //_sourceFiles.clear();
```

Comment:
If this should keep being commented out, then it needs another comment explaining why. Otherwise, remove it :)